nixos/modules/services/web-apps/nextcloud-backup.nix

```diff
@@ -0,0 +1,199 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.nextcloud;
+  backupRsyncCommand = "${pkgs.rsync}/bin/rsync -rlt --del --perms";
+  restoreRsyncCommand = "${backupRsyncCommand} --chmod=u+rwX --chown=nextcloud:nextcloud";
+  dbBackupRestore =
+    if cfg.config.dbtype == "sqlite"
+    then
+      {
+        # The sqlite database is automatically backed up since it's contained in the datadir
+        backupFileName = "data/${cfg.config.dbname}.db";
+        backupCommand = "";
+        restoreCommand = "";
+      }
+    else if cfg.config.dbtype == "pgsql"
+    then
+      let
+        backupFileName = "database-pgsql.bak";
+        pg_dump = "${config.services.postgresql.package}/bin/pg_dump";
+        pg_restore = "${config.services.postgresql.package}/bin/pg_restore";
+        psql = "${config.services.postgresql.package}/bin/psql";
+        setPassword =
+          if cfg.config.dbpassFile != null
+          then ''
+            export PGPASSWORD="$(cat ${cfg.config.dbpassFile})"
+          ''
+          else "";
+        commonFlags = "-h ${cfg.config.dbhost} -U ${cfg.config.dbuser} --no-password";
```

Comment:
The `postgresqlBackup` module doesn't allow for multiple different backups. So if someone uses it to back up some databases (other than the nextcloud db), their backup would automatically include the nextcloud database as well since we'd have to include it in `services.postgresqlBackup.databases`. Also, backing up nextcloud would force a backup of all the other databases. I don't really see a way to avoid this conflict with the current `postgresqlBackup` module.